src/kratos.ts

```diff
@@ -59,75 +60,78 @@ export const clearAuthentication = async (
   req: FastifyRequest,
   res: FastifyReply,
   reason: string,
-): Promise<void> => {
-  req.log.info(
-    {
-      reason,
-      userId: req.userId,
-      cookie: req.cookies[cookies.kratos.key],
-    },
-    'clearing authentication',
-  );
-  req.trackingId = await generateTrackingId();
-  req.userId = undefined;
-  setTrackingId(req, res, req.trackingId);
-  setCookie(req, res, 'auth', undefined);
-  setCookie(req, res, 'kratosContinuity', undefined);
-  setCookie(req, res, 'kratos', undefined);
-};
+): Promise<void> =>
+  runInSpan('clearAuthentication', async () => {
+    req.log.info(
+      {
+        reason,
+        userId: req.userId,
+        cookie: req.cookies[cookies.kratos.key],
+      },
+      'clearing authentication',
+    );
+    req.trackingId = await generateTrackingId();
+    req.userId = undefined;
+    setTrackingId(req, res, req.trackingId);
+    setCookie(req, res, 'auth', undefined);
+    setCookie(req, res, 'kratosContinuity', undefined);
+    setCookie(req, res, 'kratos', undefined);
+  });
 
 type WhoamiResponse =
   | { valid: true; userId: string; expires: Date; cookie?: string }
   | { valid: false };
 
 export const dispatchWhoami = async (
   req: FastifyRequest,
-): Promise<WhoamiResponse> => {
-  if (heimdallOrigin === 'disabled' || !req.cookies[cookies.kratos.key]) {
-    return { valid: false };
-  }
-  try {
-    const { res: whoami, headers } = await fetchKratos(
-      req,
-      `${heimdallOrigin}/api/whoami`,
-    );
-    if (whoami?.identity?.traits?.userId) {
-      return {
-        valid: true,
-        userId: whoami.identity.traits.userId,
-        expires: new Date(whoami.expires_at),
-        cookie: headers.get('set-cookie'),
-      };
+): Promise<WhoamiResponse> =>
+  runInSpan('dispatchWhoami', async () => {
```

Comment:
Since we have one for `retryFetch` this feels redundant

Reply:
Yeah, I'll remove ita